activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -560,7 +560,11 @@ def find_last(limit)
 
       def ordered_relation
         if order_values.empty? && (implicit_order_column || primary_key)
-          order(arel_attribute(implicit_order_column || primary_key).asc)
+          if implicit_order_column && implicit_order_column != primary_key
```

Comment:
```suggestion
          if implicit_order_column && primary_key && implicit_order_column != primary_key
```

If a model has no primary key and has `implicit_order_column` configured, we shouldn't add the secondary sort.

Both cases need a test.

Reply:
@eugeneius I've added tests covering the cases I could think of.